etna/records/tests/test_templatetags.py

```diff
@@ -0,0 +1,133 @@
+from django.test import SimpleTestCase, override_settings
+
+from etna.records.models import Record
+from etna.records.templatetags.records_tags import record_url
+
+
+class TestRecordURLTag(SimpleTestCase):
+
+    record_search_hit = {
+        "_source": {
+            "@template": {
+                "details": {
+                    "iaid": "e7e92a0b-3666-4fd6-9dac-9d9530b0888c",
+                    "referenceNumber": "2515/300/1",
+                    "summaryTitle": "Test",
+                }
+            }
+        },
+    }
+
+    interpretive_search_hit = {
+        "_source": {
+            "@template": {
+                "details": {
+                    "sourceUrl": "https://www.example.com",
+                    "primaryIdentifier": "bp-2304",
+                    "summaryTitle": "Test",
+                }
+            }
+        }
+    }
+
+    partial_record_dict = {
+        "iaid": "e7e92a0b-3666-4fd6-9dac-9d9530b0888c",
+        "reference_number": "2515/300/1",
+        "title": "Test",
+    }
+
+    partial_interpretive_dict = {
+        "url": "http://www.example.com",
+        "title": "Test",
+    }
+
+    record_instance = Record(
+        iaid="e7e92a0b-3666-4fd6-9dac-9d9530b0888c",
+        reference_number="2515/300/1",
+        title="Test",
+    )
+
+    record_instance_no_reference = Record(
+        iaid="e7e92a0b-3666-4fd6-9dac-9d9530b0888c",
+        reference_number="",
+        title="Test",
+    )
+
```

Comment:
`record_search_hit`, `interpretive_search_hit` gives us an eye at the API response. 
Ideally, would appreciate if the same done with these. Maybe extract the partial bit from a `record json`. _The names of those variables arn't sufficient enough to know from where the partial bit is taken from._
Alternatively, a `#comment` mentioning the path followed for the partial bit.

Reply:
> Happy to add a few comments if you think it'll help
Please do then. Thanks Andy.

Reply:
Are these sufficient @TNA-Allan? [a0de236](https://github.com/nationalarchives/ds-wagtail/pull/376/commits/a0de236c88b4961f77ccb9bbd14ad924fcb70c3c)

Reply:
`record_instance_no_reference` ... `reference_number=""`
Does API return value with empty reference?

Reply:
Well, what this is really aiming to replicate is what is currently prepared by `transform_record_result()`. Looks like the value is typically `None` when not found in the representation, so I can change this to reflect that.

Reply:
OK, please do